### ISSUE TITLE
[Bug]: v0.7.0 CSS modules styles all missing

### ISSUE THREAD

**Issue by 1yasa** [web-infra-dev/rspack#6641]
### System Info

  System:
    OS: macOS 14.4
    CPU: (32) x64 13th Gen Intel(R) Core(TM) i9-13900KF
    Memory: 10.29 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.6.1 - /usr/local/bin/node
    npm: 10.2.4 - /usr/local/bin/npm
    pnpm: 9.0.4 - /usr/local/bin/pnpm
    Watchman: 2024.05.06.00 - /usr/local/bin/watchman

### Details

All css styles missing:

<img width="1447" alt="截屏2024-05-28 18 30 01" src="https://github.com/web-infra-dev/rspack/assets/127166080/1a064ed1-bc85-4132-946b-9b6f0bd1dd33">

### Reproduce Steps

see above.

**Comment by LingyuCoder** [comment]
Could you provide a reproduction project for us to troubleshoot the issue?

**Comment by 1yasa** [comment]
> Could you provide a reproduction project for us to troubleshoot the issue?

https://github.com/1yasa/rspack_reproduce/blob/main/modules/index.css

https://github.com/1yasa/rspack_reproduce/blob/main/modules/B/index.tsx

<img width="1450" alt="截屏2024-05-28 18 45 08" src="https://github.com/web-infra-dev/rspack/assets/127166080/9002312a-a78d-488c-9cf6-f499e0e67244">

**Comment by ahabhgk** [comment]
Add `'css/auto': { namedExports: false }` to module.generator, the default css moduleType is `'css/auto'`

**Comment by 1yasa** [comment]
> Add `'css/auto': { namedExports: false }` to module.generator, the default css moduleType is `'css/auto'`

same result:

<img width="2451" alt="截屏2024-05-28 19 12 42" src="https://github.com/web-infra-dev/rspack/assets/127166080/c02c6220-aba2-438a-9c5c-313929b48ba8">

https://github.com/1yasa/rspack_reproduce/blob/main/rspack.config.ts

**Comment by 1yasa** [comment]
same for module.generator:

<img width="2518" alt="截屏2024-05-28 19 17 26" src="https://github.com/web-infra-dev/rspack/assets/127166080/808b5063-5ee7-465b-aaca-e66ca6b19de0">

**Comment by ahabhgk** [comment]
You need to add a rule to make index.css a [CSS Modules](https://github.com/css-modules/css-modules), in your case you need to add something like this:

```js
module.exports = {
  module: {
    parser: {
      'css/module': { namedExports: false } // <--
    },
    rules: [{
      test: /\.css$/,
      type: 'css/module', // <--
    }]
  }
}
```

And 'css/auto' will decide a module's moduleType by its path, path ends with `.module.css` will be treated as CSS Modules, [docs](https://arc.net/l/quote/ziljdkrx)

**Comment by 1yasa** [comment]
problem still exist in my project, config:

```ts
import { resolve } from 'path'

import { RsdoctorRspackPlugin } from '@rsdoctor/rspack-plugin'
import { defineConfig } from '@rspack/cli'
import { CopyRspackPlugin, HtmlRspackPlugin } from '@rspack/core'
import ReactRefreshPlugin from '@rspack/plugin-react-refresh'

const is_dev = process.env.NODE_ENV === 'development'
const is_prod = process.env.NODE_ENV === 'production'
const is_doctor = process.env.DOCTOR === 'true'
const is_module = false

const plugins_dev = [
	new ReactRefreshPlugin({
		exclude: [/node_modules/]
	})
]
const plugins_prod = [
	is_doctor && new RsdoctorRspackPlugin(),
	new CopyRspackPlugin({
		patterns: [{ from: './public', to: './', globOptions: { ignore: ['**/index.html'] } }]
	})
]

module.exports = defineConfig({
	devtool: is_dev ? 'source-map' : false,
	entry: {
		main: './runtime/index.tsx'
	},
	output: {
		clean: is_prod
	},
	watchOptions: {
		ignored: /node_modules/
	},
	resolve: {
		extensions: ['.tsx', '.ts', '.js'],
		tsConfigPath: resolve(__dirname, 'tsconfig.json'),
		alias: process.env.DOCTOR && {
			tslib: resolve(__dirname, 'node_modules/tslib')
		}
	},
	devServer: {
		compress: false
	},
	experiments: {
		outputModule: is_module,
		lazyCompilation: false
	},
	plugins: [
		new HtmlRspackPlugin({
			title: 'IF - GTD for prefessionals.',
			template: './public/index.html',
			scriptLoading: is_module ? 'module' : 'defer'
		}),
		...(is_dev ? plugins_dev : plugins_prod)
	],
	ignoreWarnings: [/Conflicting order/],
	module: {
		parser: {
			css: { namedExports: false },
			'css/auto': { namedExports: false },
			'css/module': { namedExports: false }
		},
		rules: [
			{
				test: /\.ts$/,
				use: {
					loader: 'builtin:swc-loader',
					options: {
						jsc: {
							parser: {
								syntax: 'typescript',
								dynamicImport: true,
								exportDefaultFrom: true,
								exportNamespaceFrom: true,
								decorators: true
							},
							transform: {
								legacyDecorator: true,
								decoratorMetadata: true
							},
							minify: {
								compress: {
									drop_console: is_prod
								}
							},
							externalHelpers: true
						},
						env: {
							targets: 'chrome >= 120'
						}
					}
				}
			},
			{
				test: /\.tsx$/,
				exclude: [/[\\/]node_modules[\\/]/],
				use: {
					loader: 'builtin:swc-loader',
					options: {
						jsc: {
							parser: {
								syntax: 'typescript',
								tsx: true,
								dynamicImport: true,
								exportDefaultFrom: true,
								exportNamespaceFrom: true,
								decorators: true
							},
							transform: {
								legacyDecorator: true,
								decoratorMetadata: true,
								react: {
									development: is_dev,
									refresh: is_dev,
									runtime: 'automatic',
									useBuiltins: true
								}
							},
							minify: {
								compress: {
									drop_console: is_prod
								}
							},
							externalHelpers: true,
							experimental: {
								plugins: [['swc-plugin-jsx-control-statements', {}]]
							}
						},
						env: {
							targets: 'chrome >= 120'
						}
					}
				}
			},
			{
				test: /\.global\.css$/,
				type: 'css',
				use: [
					{
						loader: 'postcss-loader',
						options: {
							postcssOptions: {
								plugins: ['postcss-import', 'postcss-nested', 'postcss-calc']
							}
						}
					}
				]
			},
			{
				test: /\.css$/,
				type: 'css/module',
				exclude: /\.global\.css$/,
				use: [
					{
						loader: 'postcss-loader',
						options: {
							postcssOptions: {
								plugins: ['postcss-import', 'postcss-nested', 'postcss-calc']
							}
						}
					}
				]
			},
			{
				test: /\.(png|svg|jpg)$/,
				type: 'asset/source'
			}
		]
	}
})
```

Classname correct, but styles is missing:
<img width="2876" alt="截屏2024-05-28 23 52 41" src="https://github.com/web-infra-dev/rspack/assets/127166080/c9e8b5da-d43a-4019-8acd-757b4720682f">
<img width="1334" alt="截屏2024-05-28 23 53 04" src="https://github.com/web-infra-dev/rspack/assets/127166080/fe740c56-91c3-4bec-8469-7cb7e5ae57e7">

**Comment by ahabhgk** [comment]
What's the content of index.css

**Comment by 1yasa** [comment]
> What's the content of index.css, does it using `composes`?

Nope, just simple css:

<img width="1076" alt="截屏2024-05-29 12 20 20" src="https://github.com/web-infra-dev/rspack/assets/127166080/e9f44243-0eb0-43e2-ac39-29703a9f8a0b">

**Comment by 1yasa** [comment]
@ahabhgk 

I find the reason:

<img width="2791" alt="截屏2024-05-29 12 39 14" src="https://github.com/web-infra-dev/rspack/assets/127166080/0b2f6e5a-3d0d-4f1b-9bf2-8c5ddfa086fc">

https://github.com/1yasa/rspack_reproduce/blob/main/package.json

<img width="751" alt="截屏2024-05-29 12 36 56" src="https://github.com/web-infra-dev/rspack/assets/127166080/c8c20473-074d-4144-ac13-0d4cba81c80a">

**Comment by 1yasa** [comment]
Cause by package name like this: `@openages/rspack-reproduce`

**Comment by 1yasa** [comment]
<img width="1258" alt="截屏2024-05-29 12 57 12" src="https://github.com/web-infra-dev/rspack/assets/127166080/6b4c50aa-4f5f-4b4d-8fe7-fe917260ab4b">

In v0.6.5, `@` symbol has been specially converted to '\@'."

**Comment by ahabhgk** [comment]
I see, the fix will released in v0.7.1.

Thanks for reporting this bug!
